### PR TITLE
Fixes #25038: Switch service related to compliance persistance from box to IOResult

### PR DESCRIPTION
**Pull request by fanf**
https://issues.rudder.io/issues/25038

Extremelly interesting PR with a lot of `s/Box/IOResult`

I'm commenting inline the intersting parts. 

EDIT: ok, I didn't find anything interesting at all :/ 

EDIT2: I checked in plugins, none is impacted by these changes.

### CODE REVIEW

**Comment by fanf** [comment]
PR updated with a new commit

File: webapp/sources/rudder/rudder-web/src/main/scala/com/normation/rudder/web/services/AsyncComplianceService.scala [hunk 1]
```diff
@@ -241,17 +242,4 @@ class AsyncComplianceService(
...
-    node.computeCompliance() match {
-      case Full(contentMap) =>
-        contentMap.get(nodeId) match {
-          case Some(Some(compliance)) => Full(compliance)
-          case _                      => Failure(s"No compliance found for ${nodeId.value}")
-        }
-      case eb: EmptyBox =>
-        eb ?~! s"Compute compliance of ${nodeId.value} failed"
-    }
-  }
```
Comment (clarktsiory) [inline webapp/sources/rudder/rudder-web/src/main/scala/com/normation/rudder/web/services/AsyncComplianceService.scala:256]: Interesting ! Dead code :axe:

**Review by clarktsiory — APPROVED** [review]
LGTM !

**Comment by clarktsiory** [comment]
OK, squash merging this PR

**Pull request merged**